ballerina/client.bal

```diff
@@ -0,0 +1,79 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+# Consists of APIs to integrate with Avro Schema Registry.
+public isolated class Client {
+
+    private handle schemaRegistryClient;
+
+    # Gets invoked to initialize the `connector`.
+    #
+    # + config - The configurations to be used when initializing the `connector`
+    # + return - An `cregistry:Error` if connector initialization failed
+    public isolated function init(ConnectionConfig config) returns Error? {
+        self.schemaRegistryClient = generateSchemaRegistryClient(config);
+    }
+
+    # Registers a schema with the schema registry.
+    #
+    # + subject - The subject under which the schema should be registered
+    # + schema - The Avro schema to be registered
+    # + return - The ID of the registered schema, or an `cregistry:Error` if registration fails
+    public isolated function register(string subject, string schema) returns int|Error {
+        lock {
+            return register(self.schemaRegistryClient, subject, schema);
+        }
+    }
+
+    # Retrieves a schema from the schema registry by its ID.
+    # 
+    # + id - The ID of the schema to retrieve
+    # + return - The retrieved schema or an `cregistry:Error` if the schema does not exist
+    public isolated function getById(int id) returns string|Error {
+        lock {
+            return getById(self.schemaRegistryClient, id);
+        }
+    }
+
+    # Retrieves the ID for the given subject and schema from the schema registry client.
+    # 
+    # + subject -  The subject of the schema
+    # + schema - The Avro schema
+    # + return - Returns the ID of the schema if found, or an `cregistry:Error` if an error occurs
+    public isolated function getId(string subject, string schema) returns int|Error {
+        lock {
+            return getId(self.schemaRegistryClient, subject, schema);
+        }
+    }
+}
+
+isolated function generateSchemaRegistryClient(ConnectionConfig regclient) returns handle = @java:Constructor {
+    'class: "io.ballerina.lib.confluent.CustomSchemaRegistryClient"
+} external;
+
+isolated function register(handle schemaRegistryClient, string subject, string schema) returns int|Error = @java:Method {
+    'class: "io.ballerina.lib.confluent.CustomSchemaRegistryClient"
+} external;
+
+isolated function getById(handle schemaRegistryClient, int id) returns string|Error = @java:Method {
+    'class: "io.ballerina.lib.confluent.CustomSchemaRegistryClient"
+} external;
+
+isolated function getId(handle schemaRegistryClient, string subject, string schema) returns int|Error = @java:Method {
+    'class: "io.ballerina.lib.confluent.CustomSchemaRegistryClient"
+} external;
```

Comment:
Do we need to have separate methods for extern-functions. Can't we mark the relevant functions as extern-functions itself ? Can you check this [1] 

[1] - [https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/blob/480b73e566a5888d89abb83e6df83cd80c16145e/ballerina/destination.bal#L52](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/blob/480b73e566a5888d89abb83e6df83cd80c16145e/ballerina/destination.bal#L52)